### PR TITLE
Fix OutStreams Line Plot.

### LINKED ISSUE #722: Outstreams Line Plot is not working correctly with unsync HistorySet
--------
Issue Description
--------
##### What did you expect to see happen?
The plots should be continuous. 

##### What did you see instead?
to_plot_line

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
close #722 

##### What are the significant changes in functionality due to this change request?
Fix the defect in the OutStreams line plot.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: tests/framework/OutStreams/tests [hunk 1]
```diff
@@ -58,4 +58,9 @@
   input = 'test_legend.xml'
   output = 'plot/legendTest.png'
  [../]
+ [./line]
+  type = 'RavenFramework'
+  input = 'line_plot.xml'
+  output = 'linePlot/to_plot_line.png'
```
Comment (alfoa) [inline tests/framework/OutStreams/tests]: can you add this as "image" test? at least when we activate it we will have it ready.

File: tests/framework/OutStreams/projectile_mod.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
```
Comment (alfoa) [inline tests/framework/OutStreams/projectile_mod.py]: How different is this model from the one in the analytic test?

File: tests/framework/OutStreams/line_plot.xml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" ?>
+<Simulation verbosity = 'debug'>
+  <RunInfo>
+    <WorkingDir>linePlot</WorkingDir>
+    <Sequence>generate_data,plot</Sequence>
+    <batchSize>1</batchSize>
+  </RunInfo>
+
+  <TestInfo>
```
Comment (alfoa) [inline tests/framework/OutStreams/line_plot.xml]: can you move this block to the top of the file?

File: tests/framework/OutStreams/tests (hunk 1 above)
Comment (wangcj05) [reply tests/framework/OutStreams/tests]: added

File: tests/framework/OutStreams/projectile_mod.py (hunk 2 above)
Comment (wangcj05) [reply tests/framework/OutStreams/projectile_mod.py]: The main different is the way to determine the number of time steps for each samples. This model will generate different time steps for different samples, which will output an unsync HistorySet.

File: tests/framework/OutStreams/line_plot.xml (hunk 3 above)
Comment (wangcj05) [reply tests/framework/OutStreams/line_plot.xml]: moved.

File: tests/framework/OutStreams/projectile_mod.py (hunk 2 above)
Comment (PaulTalbot-INL) [reply tests/framework/OutStreams/projectile_mod.py]: Can we add an argument to pass to "projectile" by way of constant to trigger the choice of time steps instead?

Comment (alfoa) [reply tests/framework/OutStreams/projectile_mod.py]: agree

Comment (alfoa) [reply tests/framework/OutStreams/projectile_mod.py]: any news on this?

Comment (wangcj05) [reply tests/framework/OutStreams/projectile_mod.py]: fixed.

**Comment by alfoa** [comment]
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. ***ok***
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). ***n/a***
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. ***ok***
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. ***ok***
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. ***test added***
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. ***n/a***
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. ***#722***
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? ***n/a***

**Comment by alfoa** [comment]
Check list passed...

Ready to merge...

**Comment by alfoa** [comment]
Tests passed... Merging...

**Pull request merged**